libcxx/include/__memory/scoped_temporary_buffer.h

```diff
@@ -0,0 +1,122 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCPP___MEMORY_SCOPED_TEMPORARY_BUFFER_H
+#define _LIBCPP___MEMORY_SCOPED_TEMPORARY_BUFFER_H
+
+#include <__config>
+#include <__memory/allocator.h>
+#include <__type_traits/is_constant_evaluated.h>
+#include <cstddef>
+#include <new>
+
+#if !defined(_LIBCPP_HAS_NO_PRAGMA_SYSTEM_HEADER)
+#  pragma GCC system_header
+#endif
+
+_LIBCPP_BEGIN_NAMESPACE_STD
```

Comment:
Destruction of the buffer holder needs to know the size in constant evaluation (perhaps needed for [P2562R1](https://wg21.link/p2562r1)). So in order to avoid duplication, it might be better to return a `unique_ptr<T, sized_buffer_deleter<T>>` where `sized_buffer_deleter<T>` stores the size. But I guess this won't simplify the implementation strategy a lot.

Reply:
Why not? It would avoid most of the class members we currently have. It would also give us a widely known interface instead of yet another special class for basically the same thing.